Update request user agents

bguertin opened the pull request:
Fixes https://openedx.atlassian.net/browse/MA-1833

Now user agent will always be set, and the app name/id/version will be appended.

WebView requests will have a user-agent like this:
`Mozilla/5.0 (Linux; Android 5.1; XT1053 Build/LPA23.12-15.5; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/46.0.2490.76 Mobile Safari/537.36 edX/org.edx.mobile/1.0.09`

API requests (OkHttp and Apache) will have a user-agent like this:
`Dalvik/2.1.0 (Linux; U; Android 5.1; XT1053 Build/LPA23.12-15.5) edX/org.edx.mobile/1.0.09`

bguertin commented:
@BenjiLee ?

aleffert commented:
Can you post an example to https://openedx.atlassian.net/browse/DOC-2531

bguertin commented:
> Can you post an example

Done.

Looks like: `Dalvik/2.1.0 (Linux; U; Android 5.1; XT1053 Build/LPA23.12-15.5) edX/org.edx.mobile/1.0.09`

aleffert commented:
Do you know what user agent is used for non webview request? It would probably be useful for us to document too.

bguertin commented:
@aleffert I don't think `OkHttp` sets a user agent by default.

bguertin commented:
Fixed to actually use the webview's user agent, not the app default (so we can detect webkit, chrome, etc. if we need to).

Also, added an OkHttp interceptor to set the user-agent to the app default plus our custom suffix.

So, webview requests will have a user-agent like this:
`Mozilla/5.0 (Linux; Android 5.1; XT1053 Build/LPA23.12-15.5; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/46.0.2490.76 Mobile Safari/537.36 edX/org.edx.mobile/1.0.09`

And OkHttp requests will have a user-agent like this:
`Dalvik/2.1.0 (Linux; U; Android 5.1; XT1053 Build/LPA23.12-15.5) edX/org.edx.mobile/1.0.09`

I didn't bother updating the old Apache HTTP requests, since we plan to delete it ASAP.

BenjiLee commented:
LGTM :+1:
